### PR TITLE
51 updated fully functional main page

### PR DESCRIPTION
**Pull request by ayomanetsan**
Solved the previously mentioned issues as well as updated the styles of the dropdown elements.

### CODE REVIEW

File: frontend/src/app/modules/time-table/suitable-lesson-day/suitable-lesson-day.component.ts [hunk 1]
```diff
@@ -7,62 +7,38 @@ import * as moment from 'moment';
...
-            } if (target.classList.contains('day-elem-blue-today')) {
-                target.classList.remove('day-elem-blue');
-                target.classList.remove('day-elem-blue-today');
-                target.classList.add('day-elem-white-today');
-            } else {
-                target.classList.remove('day-elem-white');
-                target.classList.add('day-elem-blue');
-                if (target.classList.contains('day-elem-blue')) {
-                    const isToday = moment(this.item).isSame(this.todayDate, 'days');
+        return isSameSeleted;
```
[the remark this answers is not in the document]
Comment (Anatolii-Ivanov) [reply frontend/src/app/modules/time-table/suitable-lesson-day/suitable-lesson-day.component.ts:37]: Everywhere

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 2]
```diff
@@ -37,41 +36,44 @@ export class SuitableLessonComponent implements OnInit {
...
 
-            promises.push(promise);
+            this.days.push({ date, meetingsAmount: 0 });
         }
 
-        const days = await Promise.all(promises);
-
-        this.days = days.sort((a, b) => a.date.getTime() - b.date.getTime());
-    }
+        const requestDate = this.selectedDate.toISOString().slice(0, 10);
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:62]: Why you make it to string...

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 3]
```diff
@@ -37,41 +36,44 @@ export class SuitableLessonComponent implements OnInit {
...
+
+        this.dateSelected.emit(new Date(this.selectedDate));
     }
 
     weekDec(): void {
         this.selectedDate.setDate(this.selectedDate.getDate() - this.amountOfDayInWeek);
 
         this.setDays();
+
+        if (!(this.selectedDate.getTime() + 10000 < new Date().getTime())) {
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:48]: And can you make it more undestandable?

Comment (Anatolii-Ivanov) [reply frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:48]: Use moment() where you need to work with date

File: frontend/src/app/shared/components/interests-dropdown/interests-dropdown.component.ts [hunk 4]
```diff
@@ -8,23 +8,23 @@ import { getTags } from '@shared/utils/tagsForInterests';
     styleUrls: ['./interests-dropdown.component.sass'],
 })
 export class InterestsDropdownComponent {
-    toggle: boolean = false;
```
Comment (KaterynaHerasymiva) [inline frontend/src/app/shared/components/interests-dropdown/interests-dropdown.component.ts:11]: Why did you remove the type?

File: frontend/src/app/shared/components/interests-dropdown/interests-dropdown.component.ts [hunk 5]
```diff
@@ -8,23 +8,23 @@ import { getTags } from '@shared/utils/tagsForInterests';
...
+    outputList = new Set<string>();
 
     selectInterest($event: Event) {
         const ev = $event.target as HTMLInputElement;
         const numb = parseInt(ev.id, 10);
         const { checked } = ev;
 
         if (checked) {
-            this.outputList = this.outputList.concat(this.inputList[numb].icon_name);
+            this.outputList.add(this.inputList[numb].icon_name);
```
Comment (KaterynaHerasymiva) [inline frontend/src/app/shared/components/interests-dropdown/interests-dropdown.component.ts:25]: @tatianahutii recommended collections to be immutable. Why did you change it?

File: frontend/src/app/shared/components/interests-dropdown/interests-dropdown.component.ts [hunk 6]
```diff
@@ -8,23 +8,23 @@ import { getTags } from '@shared/utils/tagsForInterests';
...
         } else {
-            this.outputList = this.outputList.filter(x => x !== this.inputList[numb].icon_name);
+            this.outputList.delete(this.inputList[numb].icon_name);
```
Comment (KaterynaHerasymiva) [inline frontend/src/app/shared/components/interests-dropdown/interests-dropdown.component.ts:27]: same, is it immutable?

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts [hunk 7]
```diff
@@ -37,41 +36,44 @@ export class SuitableLessonComponent implements OnInit {
...
         this.days = [];
-        const promises: Promise<{ date: Date, meetingsCount: number }>[] = [];
 
         for (let i = 0; i < 7; i++) {
             const date = moment(this.selectedDate).add(i - this.selectedDate.getDay() + 1, 'day').toDate();
-            const promise = this.getMeetingCount(date).then(meetingsCount => ({ date, meetingsCount }));
 
-            promises.push(promise);
+            this.days.push({ date, meetingsAmount: 0 });
         }
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.ts:60]: What about like
this.days = Array(7).keys().map(el => {
    const date = moment(this.selectedDate).add(el - this.selectedDate.getDay() + 1, 'day').toDate();
    return { date; meetingsAmount: 0 };
})

**Review by Anatolii-Ivanov — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/filter-section/filter-section/filter-section.component.ts [hunk 8]
```diff
@@ -38,26 +33,28 @@ export class FilterSectionComponent implements OnInit {
         this.selectedLanguageFiltersChange.emit(this.selectedLanguageFilters);
     }
 
-    removeTopic(title: string) {
-        this.selectedTopicsFilters.delete(title);
-        this.selectedTopicsFiltersChange.emit(this.selectedTopicsFilters);
+    removeInterest(title: string) {
+        this.selectedInterestsFilters.delete(title);
```
Comment (tatianahutii) [inline frontend/src/app/modules/filter-section/filter-section/filter-section.component.ts:37]: avoid using mutations

File: frontend/src/app/modules/lessons/lessons-page/lessons-page.component.html [hunk 9]
```diff
@@ -12,20 +12,6 @@ <h2 class="date">{{ todayDate }}</h2>
     </div>
 
     <div class="no-lessons" *ngIf="lessons.length === 0">
```
Comment (tatianahutii) [inline frontend/src/app/modules/lessons/lessons-page/lessons-page.component.html:14]:
```suggestion
    <div class="no-lessons" *ngIf="!lessons.length">
```

File: frontend/src/app/modules/lessons/lessons-page/lessons-page.component.ts [hunk 10]
```diff
@@ -18,19 +18,23 @@ import { LessonsCreateComponent } from '../lessons-create/lessons-create.compone
...
+    lessons: Lesson[];
 
     constructor(private dialogRef: MatDialog, private lessonService: LessonsService) { }
 
     ngOnInit(): void {
+        this.selectedDateFilter = new Date();
+
+        this.getLessons();
+
         const formatter = new Intl.DateTimeFormat('en-US', { day: 'numeric', month: 'long', year: 'numeric', weekday: 'long' });
```
Comment (tatianahutii) [inline frontend/src/app/modules/lessons/lessons-page/lessons-page.component.ts:36]: use moment for this

File: frontend/src/app/modules/lessons/lessons-page/lessons-page.component.ts [hunk 11]
```diff
@@ -93,4 +99,19 @@ export class LessonsPageComponent implements OnInit, OnChanges {
...
+
+        if (noInterests && noLanguages) {
+            return 'Oops, there are no lessons for this day. Please consider another date';
+        } if (!noInterests && !noLanguages) {
+            return 'Oops, there are no lessons with such filters for this day. Please consider another date';
+        } if (!noInterests && noLanguages) {
+            return 'Oops, there are no lessons with such interests for this day. Please consider another date';
+        }
+
+        return 'Oops, there are no lessons with such levels for this day. Please consider another date';
```
Comment (tatianahutii) [inline frontend/src/app/modules/lessons/lessons-page/lessons-page.component.ts:115]: use switch/case here

File: frontend/src/app/shared/components/interests-dropdown/interests-dropdown.component.ts (hunk 5 above)
Comment (tatianahutii) [reply frontend/src/app/shared/components/interests-dropdown/interests-dropdown.component.ts:25]: avoid using mutations

File: frontend/src/app/shared/components/interests-dropdown/interests-dropdown.component.ts (hunk 6 above)
Comment (tatianahutii) [reply frontend/src/app/shared/components/interests-dropdown/interests-dropdown.component.ts:27]: same here

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.html [hunk 12]
```diff
@@ -15,7 +15,8 @@ <h1>Find a Suitable Lesson</h1>
                     <img src="assets\calendar\calendar.svg" />
                 </div>
 
-                {{ days[0].date.toLocaleDateString() }} - {{ days[6].date.toLocaleDateString() }}
+                {{ days[0].date.toLocaleDateString('en-GB', { day: '2-digit', month: 'long' }) }} - {{ days[6].date.toLocaleDateString('en-US', { day: '2-digit', month: 'long' }) }} {{ days[0].date.getFullYear() }}
+
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.html:19]: Use date directive for this...

Comment (Anatolii-Ivanov) [reply frontend/src/app/modules/time-table/suitable-lesson/suitable-lesson.component.html:19]: https://angular.io/api/common/DatePipe

File: frontend/src/app/modules/lessons/lessons-page/lessons-page.component.ts [hunk 13]
```diff
@@ -69,12 +66,14 @@ export class LessonsPageComponent implements OnInit, OnChanges {
 
     getLessons() {
         this.lessonService.getFilteredLessons({
-            languageLevels: Array.from(this.selectedLanguageFilters).map((level: string) => Object.values(LanguageLevels).indexOf(level)),
-            tags: Array.from(this.selectedTopicsFilters).map((topic) => ({ name: topic })),
+            languageLevels: this.selectedLanguageFilters.map((level: string) => Object.values(LanguageLevels).indexOf(level)),
```
Comment (tatianahutii) [inline frontend/src/app/modules/lessons/lessons-page/lessons-page.component.ts:69]: Could we simplify this?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]

**Review by KaterynaHerasymiva — APPROVED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]
